graphql/types/SponsoredProduct.graphql

```diff
@@ -0,0 +1,74 @@
+type SponsoredProduct {
+  """
+  Unique identifier of the product. Can be SKU, EAN or VTEX Product ID.
+  """
+  identifier: ProductUniqueIdentifier
+
+  """
+  ID of the sponsored product.
+  """
+  productId: ID @deprecated(reason: "Use `identifier` instead.")
+
+  """
+  Merchandising rule applied to the product.
+  """
+  rule: Rule
+
+  """
+  Sponsored campaign related information should be added here.
+  """
+  advertisement: Advertisement
+}
+
+type ProductUniqueIdentifier {
+  """
+  Product ID value. If null, consider SKU as the default.
+  """
+  field: ProductUniqueIdentifierField
+  """
+  Product ID value.
+  """
+  value: ID
+}
+
+enum ProductUniqueIdentifierField {
+  """
+  VTEX Product ID, unique to this account.
+  """
+  id
+
+  """
+  SKU ID, unique to each variation of a product such as size, color, etc.
+  """
+  sku
+}
```

Comment:
There are more identifier fields, but I've decided not to add them, since it would imply that the `intelligent-search-api` would know how to resolve them. So, for now, I'll just keep these two.